### PR TITLE
fixed to app provider

### PR DESCRIPTION
**Pull request by CaseyBurns**
Fixed: App Provider XPath Queries
Fixed: Logic flow no longer skips update physical path
Fixed: clean path on physical path

### CODE REVIEW

**Comment by CaseyBurns** [comment]
See #144

File: providers/app.rb [hunk 1]
```diff
@@ -67,31 +67,31 @@
...
-      Chef::Log.debug("#{new_resource} application - nothing to do")
-    else
+    if(cmd != nil)
       shell_out!(cmd)
-      
-      if ((new_resource.path && is_new_path) or
-        (new_resource.application_pool && is_new_application_pool) or
-        (new_resource.enabled_protocols && is_new_enabled_protocols))
-        was_updated = true
-      end
```
Comment (EasyAsABC123) [inline providers/app.rb]: `was_updated  = true` still needs to be here

File: providers/app.rb [hunk 2]
```diff
@@ -50,10 +50,10 @@
...
     xml = cmd_current_values.stdout
     doc = Document.new(xml)
-    is_new_path = is_new_or_empty_value?(doc.root, "APP/application/@path", new_resource.path.to_s)
-    is_new_application_pool = is_new_or_empty_value?(doc.root, "APP/application/@applicationPool", new_resource.application_pool.to_s)
-    is_new_enabled_protocols = is_new_or_empty_value?(doc.root, "APP/application/@enabledProtocols", new_resource.enabled_protocols.to_s)
-    is_new_physical_path = is_new_or_empty_value?(doc.root, "APP/application/virtualDirectory/@physicalPath", new_resource.physical_path.to_s)
+    is_new_path = is_new_or_empty_value?(doc.root, "//APP/application/@path", new_resource.path.to_s)
+    is_new_application_pool = is_new_or_empty_value?(doc.root, "//APP/application/@applicationPool", new_resource.application_pool.to_s)
+    is_new_enabled_protocols = is_new_or_empty_value?(doc.root, "//APP/application/@enabledProtocols", new_resource.enabled_protocols.to_s)
+    is_new_physical_path = is_new_or_empty_value?(doc.root, "//APP/application/virtualDirectory/@physicalPath", windows_cleanpath(new_resource.physical_path.to_s))
```
Comment (EasyAsABC123) [inline providers/app.rb]: i'll run this test on a couple of boxes.

**Comment by EasyAsABC123** [comment]
@ksubrama where is the currybot when you need him?

**Comment by ksubrama** [comment]
Hi @CaseyBurns.  I tried to manually check if you had singled a CLA with Chef and it doesn't seem like SpendVision has a CCLA.  Could you follow https://supermarket.chef.io/become-a-contributor?

File: providers/app.rb (hunk 2 above)
Comment (EasyAsABC123) [inline providers/app.rb]: tested and working on windows 2012r2, anyone able to test on a lower windows?

**Comment by EasyAsABC123** [comment]
@CaseyBurns please address the above code issue with `was_updated`

**Comment by chef-supermarket** [comment]
Hi. I am an automated pull request bot named Curry. There are commits in this pull request whose authors are not yet authorized to contribute to Chef Software, Inc. projects or are using a non-GitHub verified email address. To become authorized to contribute, you will need to sign the Contributor License Agreement (CLA) as an individual or on behalf of your company. [You can read more on Chef's blog.](https://www.chef.io/blog/2014/06/23/changes-to-the-contributor-license-agreement-process)

## Non-GitHub Verified Committers

There are 1 commit author(s) whose commits are authored by a non-GitHub verified email address. Chef will have to manually verify that they are authorized to contribute.

[Please sign the CLA here.](https://supermarket.chef.io/become-a-contributor)

**Comment by ksubrama** [comment]
Should be fixed in #150.  If you have more changes, place sign in with a github account that has CLAs signed and open another pull request.

Thanks!

**Pull request closed without merging**